### PR TITLE
docs: SpriteAnimation benchmark example

### PR DESCRIPTION
**Pull request by spydon**
# Description

This adds an example so that you can test how many `SpriteAnimationComponent`s that you can add before starting to drop FPS on the platform that you are running the examples on.

It adds 100 embers per tap.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [x] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

File: examples/lib/stories/animations/benchmark_example.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  static const description = '''
+See how many SpriteAnimationComponent's your platform can handle before it
+starts to drop in FPS, this is without any sprite batching and such.
+100 animation components are added per tap.
+  ''';
+
+  final emberSize = Vector2.all(20);
+  late final TextComponent emberCounter;
+  final counterPrefix = 'Animations: ';
+  final Random random = Random();
```
Comment (luanpotter) [inline examples/lib/stories/animations/benchmark_example.dart:18]: nit: why the type annotation here?

File: examples/lib/stories/animations/benchmark_example.dart [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    final halfHeight = emberSize.y / 2;
+    addAll(
+      List.generate(
+        100,
+        (_) => Ember(
+          size: emberSize,
+          position: Vector2(
+            halfWidth + (size.x - halfWidth) * random.nextDouble(),
+            halfHeight + (size.y - halfHeight) * random.nextDouble(),
+          ),
```
Comment (luanpotter) [inline examples/lib/stories/animations/benchmark_example.dart:55]: I almost wanna add a Vector2.random() so you could do something like
```
size = emberSize + (size - emberSize) * Vector2.random()
```

**Review by luanpotter — APPROVED** [review]
LGTM

File: examples/lib/stories/animations/benchmark_example.dart (hunk 2 above)
Comment (spydon) [reply examples/lib/stories/animations/benchmark_example.dart:55]: That is much more expensive though since you'd have to create two vectors

**Pull request merged**